src/modules/setup.ts

```diff
@@ -0,0 +1,277 @@
+const resolveTSConfigPath = (builder: string) => resolvePath(root, `${builder}/tsconfig.json`)
+const resolveTSLintPath = (builder: string) => resolvePath(root, `${builder}/.eslintrc`)
+
+const typingsInfo = async (account: string, workspace: string) => {
+  const http = builderHttp(account, workspace)
+  const retryOpts = {
+    retries: 2,
+    minTimeout: 1000,
+    factor: 2,
+  }
+
+  const getTypingsInfo = async (_: any, tryCount: number) => {
+    if (tryCount > 1) {
+      log.info(`Retrying...${tryCount-1} typings info`)
+    }
+    try {
+    const res = await http.get(`/_v/builder/0/typings`)
+    return res.data.typingsInfo
+    } catch (err) {
+      const statusMessage = err.response.status ?
+        `: Status ${err.response.status}` : ''
+      log.error(`Error fetching typings info ${statusMessage} (try: ${tryCount})`)
+      throw err
+    }
+  }
+  try {
+    return retry(getTypingsInfo, retryOpts)
+  } catch (e) {
+    log.error('Unable to get typings info from vtex.builder-hub.')
+    return {}
+  }
+
+}
+
+const appTypingsURL = async (account: string, workspace: string, appName: string, appVersion: string, builder: string): Promise<string> => {
+  const appId = await resolveAppId(appName, appVersion)
+  const vendor = getVendor(appId)
+  if (isLinked({'version': appId})) {
+  return `https://${workspace}--${account}.${publicEndpoint()}/_v/private/typings/linked/v1/${appId}/${typingsPath}/${builder}`
+  }
+  return `http://${vendor}.vteximg.com.br/_v/public/typings/v1/${appId}/${typingsPath}/${builder}`
+}
+
+const appsWithTypingsURLs = async (builder: string, account: string, workspace: string, appDependencies: Record<string, any>) => {
+  const result: Record<string, any> = {}
+  for (const [appName, appVersion] of Object.entries(appDependencies)) {
+    try {
+      result[appName] = await appTypingsURL(account, workspace, appName, appVersion, builder)
+    } catch (e) {
+      log.error(`Unable to generate typings URL for ${appName}@${appVersion}.`)
+    }
+  }
+  return result
+}
+
+const runYarn = (relativePath: string) => {
+  log.info(`Running yarn in ${relativePath}`)
+  execSync(
+    `${yarnPath} --force --non-interactive`,
+    {stdio: 'inherit', cwd: resolvePath(root, `${relativePath}`)}
+  )
+  log.info('Finished running yarn')
+}
+
+const yarnAddESLint = (relativePath: string) => {
+  log.info(`Adding lint configs in ${relativePath}`)
+  const lintDeps = join(' ', values(mapObjIndexed((version, name) => `${name}@${version}`, addToPackageJson)))
+  execSync( `${yarnPath} add ${lintDeps} --dev`,
+    {stdio: 'inherit', cwd: resolvePath(root, `${relativePath}`)}
+  )
+}
+
+const getTypings = async (manifest: Manifest, account: string, workspace: string) => {
+  const typingsData = await typingsInfo(account, workspace)
+
+  const buildersWithInjectedDeps =
+    pipe(
+      prop('builders'),
+      mapObjIndexed(
+        (version: string, builder: string) => {
```

Comment:
this could be a named function outside getTypings, would be easier to read.

Reply:
thx